### PR TITLE
Migrates to sender and receiver propagation handlers

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
Related PRs

- [x] https://github.com/micrometer-metrics/micrometer/pull/3293
- [x] https://github.com/micrometer-metrics/tracing/pull/36

### CODE REVIEW

**Comment by wilkinsona** [comment]
To avoid being on snapshots for too long, we should merge this nearer the release of the next milestone.

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/MicrometerTracingAutoConfiguration.java [hunk 1]
```diff
@@ -70,20 +69,20 @@ public DefaultTracingObservationHandler defaultTracingObservationHandler(Tracer
...
 	}
 
 	@Bean
 	@ConditionalOnMissingBean
-	@ConditionalOnBean({ Tracer.class, HttpClientHandler.class })
-	@Order(HTTP_CLIENT_TRACING_OBSERVATION_HANDLER_ORDER)
-	public HttpClientTracingObservationHandler httpClientTracingObservationHandler(Tracer tracer,
-			HttpClientHandler httpClientHandler) {
-		return new HttpClientTracingObservationHandler(tracer, httpClientHandler);
+	@ConditionalOnBean({ Tracer.class, Propagator.class })
```
Comment (jonatan-ivanov) [inline spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/MicrometerTracingAutoConfiguration.java:81]: This means that both receiver and sender handlers will be created if there is a propagator.

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/tracing/MicrometerTracingAutoConfigurationTests.java [hunk 2]
```diff
@@ -167,13 +167,13 @@ DefaultTracingObservationHandler customDefaultTracingObservationHandler() {
 		}
 
 		@Bean
-		HttpServerTracingObservationHandler customHttpServerTracingObservationHandler() {
-			return mock(HttpServerTracingObservationHandler.class);
+		PropagatingReceiverTracingObservationHandler customPropagatingReceiverTracingObservationHandler() {
```
Comment (jonatan-ivanov) [inline spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/tracing/MicrometerTracingAutoConfigurationTests.java:170]: GH does not let me comment it, but a few lines above the `HttpServerHandler` and `HttpClientHandler` beans are not used but a `Propagator` bean is missing.

**Review by jonatan-ivanov — CHANGES_REQUESTED** [review]

**Comment by wilkinsona** [comment]
These changes were also included in https://github.com/spring-projects/spring-boot/pull/32048 which has now been merged.

**Pull request closed without merging**